Plugin/src/main/java/xyz/kyngs/librelogin/common/listener/AuthenticListeners.java

```diff
@@ -72,7 +70,8 @@ protected void onPlayerDisconnect(P player) {
     }
 
     protected PreLoginResult onPreLogin(String username, InetAddress address) {
-        if (username.length() > 16 || !NAME_PATTERN.matcher(username).matches()) {
+        Pattern namePattern = Pattern.compile(plugin.getConfiguration().get(ConfigurationKeys.ALLOWED_NICKNAME_CHARACTERS));
```

Comment:
Pleas pre-compile the pattern and reuse it. Compiling it in each preLogin is wasteful.